Fixture Battery Proposal

Verschwiegener opened the pull request:
#171

Verschwiegener committed:
Battery Definition #171

Verschwiegener committed:
Fixed Spelling

Verschwiegener committed:
Fixed Spelling

Verschwiegener committed:
Clearer Battery Description

Verschwiegener commented:
Needs Input on how to Map between Battery Runtime and Light output

petrvanekrobe commented:
We will need to be a bit more specific, as many (not only Robe) fixtures have a battery (we will also have to distinguish between a battery and an accumulator (a rechargeable battery) which is used to power the display but not to provide power for the light output.

Verschwiegener committed:
Battery Power Modes of Operation

Verschwiegener commented:
Yeah that's a good point. Though i thought that the Wiring Object Node was more for External Component that could be wired into the fixture.

petrvanekrobe commented:
With the battery being a wiring object, pin patch could be defined. The question is whether applications are interested in that. If not, we could define it not as wiring object, inside the  Properties collect. 

Below i am tagging people/applications i am aware of which could be interested in that. We can decide based on their feedback/requirements.

@klinzey - Vectorworks
@moritzstaffel - Production Assist
@bartvanstiphout - PatchPrinter

klinzey commented:
To me if it's internal to the device it would not be a wiring object. For me a wiring object is something that needs to be connected by the user. I asked the question of our connectCAD expert.

petrvanekrobe commented:
Agree, but there are nuances...:

So this one is internal, right?

![image](https://github.com/user-attachments/assets/67853d86-28d9-4270-b477-aa1008ae3489)

[link](https://www.robe.cz/liteware-ho2)

And how about this one? One base, multiple attachments:

![image](https://github.com/user-attachments/assets/638053fc-e754-4b4e-81cd-88d895a1b275)

[link](https://www.robe.cz/liteware-satellite2)

moritzstaffel commented:
We also have Fuses that are internal an internally connected. A Geometry seems to be the right thing.

Verschwiegener commented:
Should the Battery be a WiringObject or a new BatteryObject(?) in the Geometry?

moritzstaffel commented:
A Wiring Object seems more like the other, but the all the other Power Wire Geometries are for AC, while batteries are for DC. 
So there is some challenge to connect them.

Verschwiegener commented:
Other than that i think the Name "WiringObject" is much more associated with Wires that can be connected to the Fixture and Battery's would almost always be intern. So i think a new Geometry Object would be easier to understand for the user.
But that's just my opinion

Verschwiegener commented:
When a Fixture has multiple Battery's that function as one should this also be specified?

klinzey commented:
If the batteries are internal I think it should just be represented by a single battery. 1 or multiple batteries does not effect the functionality. I think we just care about the total capacity of the battery as it may effect run time.
I think if the batteries are external then they should be represented as individual batteries.
The other issue with multiple external batteries is that I think the wiring objects expects that there is a single power supply for an object and the distributor assumes only 2 inputs can't connected to the same output.

Verschwiegener commented:
Just a bit of brainstorming

New WiringObject ComponentType "Battery"

New Attributes:
BatteryType: Soldered, Replacable, QuickSwap(?), External(?)
BatteryMode: Configuration, Operation
BatteryRuntime: Minimum Fixture Runtime on Battery Power Unit: Hours
ChargingTime: Charging Time of the Battery Unit: Hours

Changed Attribute Spec:
VoltageRangeMax: Voltage of the fully charged the Battery supplies
VoltageRangeMin: Minimum Battery Voltage the fixture functions with

I think that the Battery capacity would be irrelevant and should not be added because we would need to know the power consumption of the whole fixture in its different states to estimate the Runtime and at this point it is just easier to define a minimum Battery Runtime.

Verschwiegener committed:
Moved Battery Definition to WiringObject

petrvanekrobe commented:
I am still not clear on the use-case here, which will partially explain/dictate the requirements and what the application needs to know and what for. I would also like to know what am i describing - am i describing the battery inside the box or am i describing the box itself? Again, example: 

https://www.robe.cz/liteware-satellite2
https://astera-led.com/products/runtimeextender/

Btw, @Verschwiegener are you planning to be at LDI by any chance?

Verschwiegener commented:
For me the use-case would be to better model some gigs that i have that heavily utilize Battery Powered Fixtures. I would suggest  that unless the BatteryType is External the Battery inside the Box would be defined, like with the Liteware Satellite 2. And with something like the Runtime Extender i would say that the BatteryType is External and because the Battery is not always the same the BatteryRuntime and ChargingTime Attributes would be left empty. 

I'm not planing to go to LDI but depending on my University lectures I'm going to be at PLS.

Verschwiegener commented:
I talked to a friend who works a lot with Astera Tubes, he said that the UN number would maybe also a good addition

Verschwiegener committed:
Add UN Number

petrvanekrobe committed:
Battery proposal: remove bool

Generally, we try to avoid bool as often, over time more states then 0/1 are needed.

petrvanekrobe committed:
Battery specification - add more fields

Added new battery-related attributes to the specification.

petrvanekrobe commented:
This PR has been replaced by https://github.com/mvrdevelopment/spec/pull/311/ , where Battery is now it's own geometry.
